### ISSUE TITLE
Issues with Properties with '.' in them

### ISSUE THREAD

**Issue by vbfischer** [googleapis/nodejs-datastore#580]
TLDR; We have some existing properties that have a '.' in them. They are part of the property name, not delimitating fields in an embedded entity.

For example:
```
"tests.results": "F,F,F,F",
"tests.notes": "Some notes here"
```

For the most part this works fine, but when attempting to exclude properties for indexes, its not finding them appropriately. For example, attempting to save the following entity:

```
const updatedEntity = {
  key: datastore.key('company'),
  excludeFromIndexes: [
    "test.notes"
  ],
  data: {
    field1: "1234",
   "test.notes": "some notes that exceed 1500bytes causing indexing error"
  }
}
```

I can work around this by using the alternate format when saving:

```
const updatedEntity = {
  key: datastore.key('company'),
  data: [
  {
    name: 'field1',
    value: '1234'
  },
  {
    name: 'test.notes',
    value: "some notes that exceed 1500bytes causing indexing error",
    excludeFromIndexes: true
  }
 ] 
}
```

I'm fine with the latter format, but the source in request.ts seems to imply that it will be removed in future versions:

```
      // @TODO remove in @google-cloud/datastore@2.0.0
        // This was replaced with a more efficient mechanism in the top-level
        // `excludeFromIndexes` option.
```
#### Environment details

  - `@google-cloud/datastore` version: 5.0.3

#### Steps to reproduce

  1. Have entity with property name that has '.' in it.
  2. Specify this field from excludeFromIndexes
  3. attempt to save entity with field exceeding the 1500 bytes

Java API doesn't seem to have this issue.

**Comment by vbfischer** [comment]
I've written a simple unit test if that'd help...

**Comment by vishald123** [comment]
Hi @vbfischer, It is not recommended to use dot in property names. Please refer the documentation [here](https://cloud.google.com/datastore/docs/best-practices#properties). Also, you can use the alternate format. You can see [here](https://cloud.google.com/datastore/docs/concepts/indexes#unindexed_properties)

**Comment by crwilcox** [comment]
I am closing this as, per @vishald123 comment, as using dots in names interferes directly with the indexing.

@vbfischer if you have not been able to work through this in your codebase, or need further clarification, please reach out.

**Comment by jamie-pate** [comment]
Ironically the report is trying to exclude their objects from indexing... IE the value they want to store does not get indexed and is treated as pure data.
